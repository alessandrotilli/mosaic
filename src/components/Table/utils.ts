export const COLUMN_CHECKBOX_PATH = "checkbox-selection";
export const COLUMN_ROW_ACTIONS_PATH = "row-actions";

export const TOOLBAR_HEIGHT = 64;
export const TOOLBAR_HEIGHT_MOBILE = 56;

export const PAGINATION_TOOLBAR_HEIGHT = 64;
