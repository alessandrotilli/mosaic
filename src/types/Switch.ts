import { IInputBoolean } from "./InputBoolean";

/**
 * @deprecated Use "small" | "medium"
 */
export enum SwitchSize {
  small = "small",
  default = "medium",
}

export interface ISwitch extends IInputBoolean {
  /**
   * Disable ripple effect
   */
  disableRipple?: boolean;
}
