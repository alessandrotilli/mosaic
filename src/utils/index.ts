import { ChangeEvent, MouseEvent } from "react";

export interface ISubpart {
  label: string;
  value?: Function;
}

export interface ISubpartSuffix {
  label: string;
  suffix: string;
}

export type ISubpartMap<T extends string = string> = {
  [key in T]: ISubpart;
};

export const DATA_CY_SUFFIX_SEPARATOR = "-";

/**
 * Gets composed dataCy for components with subparts
 * @param dataCy the dataCy prefix
 * @param subpart the subpart to consider
 * @param args optional arguments for subpart evaluation
 */
export const getComposedDataCy = (dataCy: string, subpart: ISubpart, args?: any): string => {
  const { label, value } = subpart;
  return `${dataCy}${DATA_CY_SUFFIX_SEPARATOR}${!value ? slugify(label) : value(args)}`;
};

/**
 * Gets all composed dataCy for all listed subparts
 * @param subpartMap the map of all subparts to consider
 */
export const getAllComposedDataCy = (subpartMap: ISubpartMap): ISubpartSuffix[] =>
  Object.values(subpartMap).map(({ label, value }) => ({
    label,
    suffix: !value ? slugify(label) : value(),
  }));

/**
 * Gets an object property by path, if path is nested is executed recursively
 * @param object the source object
 * @param path the desired path
 */
export const getObjectProperty = (object: any, path: string): any => {
  if (!object) {
    return undefined;
  }

  if (!path.includes(".")) {
    return object[path];
  }

  const pathParts = path.split(".");
  const basePath = pathParts[0];
  if (!object[basePath]) {
    return undefined;
  }

  const baseObject = object[basePath];
  return getObjectProperty(baseObject, path.replace(`${basePath}.`, ""));
};

/**
 * Creates slug or kebab-case string from value
 * @param value the value to slugify
 */
export const slugify = (value: string) =>
  value
    .toLowerCase()
    .replace(/\s+/g, "-")
    .replace(/[^\w-]+/g, "");

/**
 * Suppresses default behaviour and propagation of a given event
 * @param event the event to suppress
 */
export const suppressEvent = (event: ChangeEvent<any> | MouseEvent<any, MouseEvent> | null) => {
  if (!event) {
    return;
  }

  event.preventDefault();
  event.stopPropagation();
};
