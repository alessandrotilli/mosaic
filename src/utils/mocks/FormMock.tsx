import React, { cloneElement, FC, PropsWithChildren, ReactElement, useMemo, useState } from "react";
import { DecoratorFn } from "@storybook/react";

type IFormValue = boolean | number | string | null;

interface FormMockType {
  onChangePropName?: string;
  value: IFormValue;
  valuePropName?: string;
}

const FormMock: FC<PropsWithChildren<FormMockType>> = ({
  children,
  onChangePropName = "onChange",
  value: externalValue,
  valuePropName = "value",
}) => {
  const [value, setValue] = useState(externalValue);

  const wrappedFormElement = useMemo(
    () =>
      cloneElement(children as ReactElement<any>, {
        [onChangePropName]: setValue,
        [valuePropName]: value,
      }),
    [children, onChangePropName, value, valuePropName]
  );

  return wrappedFormElement;
};

const formDecorator: DecoratorFn = (Story, { args }) => <FormMock value={args.value}>{Story()}</FormMock>;

export { formDecorator, FormMock };
