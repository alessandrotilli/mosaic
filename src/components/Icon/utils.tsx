import React from "react";
import MUIAccountCircleIcon from "@material-ui/icons/AccountCircle";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIAppsIcon from "@material-ui/icons/Apps";
import MUIArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import MUIBlockIcon from "@material-ui/icons/Block";
import MUIBrightness4Icon from "@material-ui/icons/Brightness4";
import MUIBusinessIcon from "@material-ui/icons/Business";
import MUICheckIcon from "@material-ui/icons/Check";
import MUICheckBoxIcon from "@material-ui/icons/CheckBox";
import MUICheckBoxOutlineBlankIcon from "@material-ui/icons/CheckBoxOutlineBlank";
import MUIChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import MUIChevronRightIcon from "@material-ui/icons/ChevronRight";
import MUICloseIcon from "@material-ui/icons/Close";
import MUICropIcon from "@material-ui/icons/Crop";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIDetailsIcon from "@material-ui/icons/Details";
import MUIDraftsIcon from "@material-ui/icons/Drafts";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIErrorIcon from "@material-ui/icons/Error";
import MUIFilterListIcon from "@material-ui/icons/FilterList";
import MUIFirstPageIcon from "@material-ui/icons/FirstPage";
import MUIGroupWorkIcon from "@material-ui/icons/GroupWork";
import MUIHomeIcon from "@material-ui/icons/Home";
import MUIKeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
import MUIKeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
import MUILanguageIcon from "@material-ui/icons/Language";
import MUILastPageIcon from "@material-ui/icons/LastPage";
import MUIListIcon from "@material-ui/icons/List";
import MUIMailIcon from "@material-ui/icons/Mail";
import MUIMenuIcon from "@material-ui/icons/Menu";
import MUINotificationsIcon from "@material-ui/icons/Notifications";
import MUIOpenInNew from "@material-ui/icons/OpenInNew";
import MUIPauseIcon from "@material-ui/icons/Pause";
import MUIPlayArrowIcon from "@material-ui/icons/PlayArrow";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUIPowerSettingsNewIcon from "@material-ui/icons/PowerSettingsNew";
import MUIPrintIcon from "@material-ui/icons/Print";
import MUIRefreshIcon from "@material-ui/icons/Refresh";
import MUISaveIcon from "@material-ui/icons/Save";
import MUIScheduleIcon from "@material-ui/icons/Schedule";
import MUISearchIcon from "@material-ui/icons/Search";
import MUISendIcon from "@material-ui/icons/Send";
import MUISettingsIcon from "@material-ui/icons/Settings";
import MUISignalWifi1BarOutlinedIcon from "@material-ui/icons/SignalWifi1BarOutlined";
import MUITimelineIcon from "@material-ui/icons/Timeline";
import MUIVisibilityIcon from "@material-ui/icons/Visibility";
import MUIVisibilityOffIcon from "@material-ui/icons/VisibilityOff";
import MUIWarningIcon from "@material-ui/icons/Warning";
import MUIWifiOffOutlinedIcon from "@material-ui/icons/WifiOffOutlined";
import MUIWifiOutlinedIcon from "@material-ui/icons/WifiOutlined";

import { Icons, IIconsCatalog } from "../../types/Icon";

export const iconsCatalog: IIconsCatalog = {
  [Icons.account]: <MUIAccountCircleIcon />,
  [Icons.add]: <MUIAddIcon />,
  [Icons.apps]: <MUIAppsIcon />,
  [Icons.block]: <MUIBlockIcon />,
  [Icons.brightness]: <MUIBrightness4Icon />,
  [Icons.business]: <MUIBusinessIcon />,
  [Icons.chart_timeline]: <MUITimelineIcon />,
  [Icons.check]: <MUICheckIcon />,
  [Icons.checkbox]: <MUICheckBoxIcon />,
  [Icons.checkbox_empty]: <MUICheckBoxOutlineBlankIcon />,
  [Icons.clock]: <MUIScheduleIcon />,
  [Icons.close]: <MUICloseIcon />,
  [Icons.connection_slow]: <MUISignalWifi1BarOutlinedIcon />,
  [Icons.crop]: <MUICropIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.details]: <MUIDetailsIcon />,
  [Icons.down]: <MUIKeyboardArrowDownIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.error]: <MUIErrorIcon />,
  [Icons.filter]: <MUIFilterListIcon />,
  [Icons.first]: <MUIFirstPageIcon />,
  [Icons.group]: <MUIGroupWorkIcon />,
  [Icons.home]: <MUIHomeIcon />,
  [Icons.language]: <MUILanguageIcon />,
  [Icons.last]: <MUILastPageIcon />,
  [Icons.left]: <MUIChevronLeftIcon />,
  [Icons.list_check]: <MUIPlaylistAddCheckIcon />,
  [Icons.list]: <MUIListIcon />,
  [Icons.mail_open]: <MUIDraftsIcon />,
  [Icons.mail]: <MUIMailIcon />,
  [Icons.menu]: <MUIMenuIcon />,
  [Icons.notifications]: <MUINotificationsIcon />,
  [Icons.open_new]: <MUIOpenInNew />,
  [Icons.pause]: <MUIPauseIcon />,
  [Icons.play]: <MUIPlayArrowIcon />,
  [Icons.power]: <MUIPowerSettingsNewIcon />,
  [Icons.print]: <MUIPrintIcon />,
  [Icons.refresh]: <MUIRefreshIcon />,
  [Icons.right]: <MUIChevronRightIcon />,
  [Icons.save]: <MUISaveIcon />,
  [Icons.search]: <MUISearchIcon />,
  [Icons.send]: <MUISendIcon />,
  [Icons.settings]: <MUISettingsIcon />,
  [Icons.up_arrow]: <MUIArrowUpwardIcon />,
  [Icons.up]: <MUIKeyboardArrowUpIcon />,
  [Icons.visibility]: <MUIVisibilityIcon />,
  [Icons.visibility_off]: <MUIVisibilityOffIcon />,
  [Icons.warn]: <MUIWarningIcon />,
  [Icons.wifi_off]: <MUIWifiOffOutlinedIcon />,
  [Icons.wifi_on]: <MUIWifiOutlinedIcon />,
};
