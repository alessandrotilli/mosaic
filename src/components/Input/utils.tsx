import React from "react";
import { InputAdornment as MUIInputAdornment } from "@material-ui/core";

import { IInputAdornment } from "../../types/Input";
import IconButton from "../IconButton";

export const getAdornment = (adornment?: IInputAdornment) => {
  if (!adornment) {
    return;
  }

  const { icon, onClick } = adornment;
  return (
    <MUIInputAdornment position="end">
      <IconButton disabled={!onClick} icon={icon} onClick={onClick!} size="small" />
    </MUIInputAdornment>
  );
};
