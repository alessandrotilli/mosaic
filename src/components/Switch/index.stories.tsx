import React from "react";
import { action } from "@storybook/addon-actions";
import { boolean, select, text } from "@storybook/addon-knobs";

import { SwitchSize } from "../../types/Switch";
import { getAllComposedDataCy } from "../../utils";
import FormMock from "../../utils/mocks/FormMock";
import IntlProviderMock, { LocaleMock, MessageMock } from "../../utils/mocks/IntlProviderMock";
import { getDocumentationPage, StoriesWrapper } from "../../utils/stories";

import Switch, { DATA_CY_DEFAULT, DATA_CY_SHORTCUT, LOCALIZABLE_PROPS, SUBPARTS_MAP } from ".";

export default {
  title: "Switch",
  component: Switch,
  parameters: {
    ...getDocumentationPage({
      basedOn: "@material-ui/core/Switch",
      component: "Switch",
      e2eTestInfo: {
        dataCyDefault: DATA_CY_DEFAULT,
        dataCyShortcut: DATA_CY_SHORTCUT,
        subpartsSuffixes: getAllComposedDataCy(SUBPARTS_MAP),
      },
      localizableProps: LOCALIZABLE_PROPS,
    }),
  },
};

export const Canvas = () => (
  // FormMock simulates external form component handling state
  // In a real case scenario "onChange" and "value" props must be passed to Switch
  <FormMock inputValue={boolean("value", true)} onInputChange={action("Change switch")}>
    <Switch
      dataCy={text("data-cy", "switch")}
      label={text("label", "Switch")}
      labelPlacement={select("labelPlacement", ["end", "start"], "start")}
      onChange={action("Change switch")}
      value={boolean("value", true)}
    />
  </FormMock>
);

export const CustomStyle = () => <Switch style={{ color: "red" }} value />;

export const Disabled = () => (
  <StoriesWrapper>
    <Switch disabled />
    <Switch value disabled />
  </StoriesWrapper>
);

export const Label = () => (
  <StoriesWrapper>
    <Switch label="Switch" />
    <Switch label="Switch" labelPlacement="end" value />
  </StoriesWrapper>
);

export const Localized = () => (
  // IntlProviderMock simulates external IntlProvider context
  <IntlProviderMock locale={select("locale", LocaleMock, LocaleMock.en)}>
    <Switch label={MessageMock.switch} localized />
  </IntlProviderMock>
);

export const Size = () => (
  <StoriesWrapper>
    <Switch value />
    <Switch value size={SwitchSize.small} />
    <Switch />
    <Switch size={SwitchSize.small} />
  </StoriesWrapper>
);

export const Values = () => (
  <StoriesWrapper>
    <Switch onChange={() => {}} />
    <Switch value onChange={() => {}} />
  </StoriesWrapper>
);
