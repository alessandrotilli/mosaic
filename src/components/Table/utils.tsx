import React, { forwardRef, Fragment } from "react";
import { Action as MTActionType, Column as MTColumnType, Options as MTOptionsType } from "material-table";
import { Icons, IconSize } from "../../types/Icon";
import { TableActionPosition, ITableAction, ITableColumn } from "../../types/Table";
import Button from "../Button";
import Icon from "../Icon";
import IconButton from "../IconButton";
import { getDataCyForSubComponent, suppressEvent } from "../../utils";

// For default props refer to Options section here https://material-table.com/#/docs/all-props
export const DEFAULT_TABLE_OPTIONS: MTOptionsType = {
  debounceInterval: 500,
  draggable: false,
  emptyRowsWhenPaging: false,
};

export const actionAdapter = (action: ITableAction): MTActionType<object> => {
  const {
    callback,
    label,
    disabled = false,
    hidden = false,
    icon = undefined,
    position = TableActionPosition.default,
  } = action;
  return {
    disabled,
    hidden,
    icon: `${label}$${icon}`,
    onClick: (event, data) => {
      if (event) {
        suppressEvent(event);
      }
      callback(data);
    },
    position,
  };
};

export const actionComponentAdapter = (props: any, dataCy: string) => {
  const { data, action } = props;
  const { disabled, hidden, icon, position, onClick }: MTActionType<object> = action;
  if (hidden) {
    return null;
  }

  const displaysButton = position === TableActionPosition.default;
  const labelAndIconName = icon as string;
  const label = labelAndIconName.split("$")[0];
  const iconName = labelAndIconName.split("$")[1] as Icons;
  return (
    <Fragment>
      {displaysButton && (
        <Fragment>
          <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
          <Button
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={!iconName ? undefined : { name: iconName }}
            label={label}
            onClick={() => onClick(undefined, data)}
          />
        </Fragment>
      )}
      {!displaysButton && (
        <Fragment>
          <IconButton
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={iconName}
            onClick={() => onClick(undefined, data)}
            size={IconSize.small}
          />
          <span>&nbsp;&nbsp;</span>
        </Fragment>
      )}
    </Fragment>
  );
};

export const columnAdapter = (column: ITableColumn): MTColumnType<object> => {
  const { label, path, render, width } = column;
  return {
    ...(!!width && ({ width } as object)),
    field: path,
    render,
    title: label,
  };
};

export const iconAdapter = (name: Icons, size = IconSize.default) =>
  forwardRef((props: any, ref: any) => {
    return <Icon forwarded={{ ...props, ref }} name={name} size={size} />;
  });
