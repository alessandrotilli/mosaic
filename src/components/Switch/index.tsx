/* eslint-disable @typescript-eslint/no-explicit-any */
import React, { ChangeEvent, FC, useCallback } from "react";
import { FormControlLabel as MUIFormControlLabel, Switch as MUISwitch } from "@mui/material";

import { ISwitch } from "../../types/Switch";
import { getComposedDataCy } from "../../utils";
import localized, { ILocalizableProperty } from "../../utils/hocs/localized";

export const DATA_CY_DEFAULT = "switch";
export const DATA_CY_SHORTCUT = "label";
export const LOCALIZABLE_PROPS: ILocalizableProperty[] = [{ name: "label", type: "string" }];

export const SUBPARTS_MAP = {
  input: {
    label: "Input",
  },
  toggle: {
    label: "Toggle",
  },
};

const Switch: FC<ISwitch> = ({
  dataCy = DATA_CY_DEFAULT,
  disabled = false,
  disableRipple = false,
  label,
  labelPlacement = "start",
  onChange,
  required = false,
  size = "medium",
  style,
  value = false,
}) => {
  const onChangeHandler = useCallback(
    ({ target: { checked } }: ChangeEvent<HTMLInputElement>) => onChange && onChange(checked),
    [onChange]
  );

  return (
    <MUIFormControlLabel
      control={
        <MUISwitch
          checked={value}
          data-cy={getComposedDataCy(dataCy, SUBPARTS_MAP.toggle)}
          disabled={disabled}
          disableRipple={disableRipple}
          inputProps={
            {
              "data-cy": getComposedDataCy(dataCy, SUBPARTS_MAP.input),
            } as any
          }
          onChange={onChangeHandler}
          required={required}
          size={size}
          style={style}
        />
      }
      data-cy={dataCy}
      label={label}
      labelPlacement={labelPlacement}
    />
  );
};

export const LocalizedSwitch = localized(Switch, {
  dataCyShortcut: DATA_CY_SHORTCUT,
  localizableProps: LOCALIZABLE_PROPS,
});

export default LocalizedSwitch;
