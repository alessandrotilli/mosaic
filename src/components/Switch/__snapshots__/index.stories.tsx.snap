// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs/Switch DisableRipple smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1f2kvjf-MuiFormControlLabel-root"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          data-cy="switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb">
      </span>
    </span>
    <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
    </span>
  </span>
</label>
`;

exports[`Inputs/Switch Disabled smoke-test 1`] = `
<label class="MuiFormControlLabel-root Mui-disabled MuiFormControlLabel-labelPlacementStart css-1f2kvjf-MuiFormControlLabel-root"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-disabled Mui-disabled PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-disabled Mui-checked Mui-disabled css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          tabindex="-1"
          aria-disabled="true"
          data-cy="switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
             disabled
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb">
      </span>
    </span>
    <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
    </span>
  </span>
</label>
`;

exports[`Inputs/Switch Label smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1f2kvjf-MuiFormControlLabel-root"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          data-cy="switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb">
      </span>
      <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
      </span>
    </span>
    <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
    </span>
  </span>
  <span class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root">
    Switch
  </span>
</label>
`;

exports[`Inputs/Switch LabelEnd smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          data-cy="switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb">
      </span>
      <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
      </span>
    </span>
    <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
    </span>
  </span>
  <span class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root">
    Switch
  </span>
</label>
`;

exports[`Inputs/Switch Localized smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1f2kvjf-MuiFormControlLabel-root"
       data-cy="locale.switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          data-cy="locale.switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
             type="checkbox"
             data-cy="locale.switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb">
      </span>
      <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
      </span>
    </span>
    <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
    </span>
  </span>
  <span class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root">
    Interruttore
  </span>
</label>
`;

exports[`Inputs/Switch Primary smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1f2kvjf-MuiFormControlLabel-root"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          data-cy="switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb">
      </span>
      <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
      </span>
    </span>
    <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
    </span>
  </span>
</label>
`;

exports[`Inputs/Switch SizeSmall smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1f2kvjf-MuiFormControlLabel-root"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeSmall css-1ycxvqq-MuiSwitch-root">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          data-cy="switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb">
      </span>
      <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
      </span>
    </span>
    <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
    </span>
  </span>
</label>
`;

exports[`Inputs/Switch Styled smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1f2kvjf-MuiFormControlLabel-root"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          data-cy="switch-toggle"
          style="color: red;"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb">
      </span>
      <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
      </span>
    </span>
    <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
    </span>
  </span>
</label>
`;
